### ISSUE TITLE
Motor control NodeMCU drops out from WiFi

### ISSUE THREAD

**Issue by frederikheld** [frederikheld/robotrain#25]
If the train is powered for a certain time, the NodeMCU is not available via MQTT anymore as it drops out form WiFi. This can happen after an hour, after a couple of hours, sometimes not at all through the whole day.

## Possible Reasons

Arduino Sketch has memory leaks. I read about that in https://github.com/knolleary/pubsubclient.
- [ ] check if the dropout is related to the number of MQTT messages sent to the NodeMCU.
> I would appreciate help on that as I'm not firm with memory management in C

NodeMCU is overheated
- [ ] with all the electronis inside, the train gets pretty warm. Active ventilation might fix the issue.

**Comment by Cjain181299** [comment]
Hey did your issue got resolved ?
I'm experiencing same thing, yesterday the connection was good but today within 1/2 hour I lost connection with ESP8266 via mqtt and had to switch it OFF and ON to get it working again.

**Comment by frederikheld** [comment]
No, not yet unfortunately. I didn't dig deeper on that one yet. If you find a solution, I'd be happy about a comment or PR ;-)

**Comment by Chetan-99** [comment]
Hey, the issue is resolved.
So I made a function for reconnecting to a wifi like : 
reconnectWifi() -> normal function that we initialize in setup()

Now in loo(), apply a check for the wifi connection: 
if ( WiFi.status() != WL_CONNECTED){
       reconnectWifi();
}

Working flawlessly. Never experienced disconnection after this.
Good Luck

**Comment by frederikheld** [comment]
Awesome, thank you very much for sharing with me! I'll look into it as soon as I am working on this project again :-) This should fix the issue for me as well :-)
